src/source-netmap.c

```diff
@@ -294,19 +345,40 @@ static int NetmapOpen(NetmapIfaceSettings *ns,
     const char *opt_x = "x"; // not for IPS
     const char *opt_z = "z"; // zero copy, not for IPS
 
-    // FreeBSD 11 doesn't have R and T.
-#if NETMAP_API<=11
-    opt_R = "";
-    opt_T = "";
-#endif
     /* assemble options string */
     char optstr[16];
     if (ns->ips)
         opt_x = "";
-// z seems to not play well with multiple opens of a real dev on linux
-//    if (!zerocopy || ips)
+    // z seems to not play well with multiple opens of a real dev on linux
     opt_z = "";
 
+    /*
+     * How netmap endpoint names are selected:
+     *
+     * The following logic within the "retry" loop builds endpoint names.
+     *
+     * IPS Mode:
+     * There are two endpoints: one hardware NIC and either a hardware NIC or host stack "NIC".
+     *
+     * IDS Mode:
+     * One endpoint -- usually a hardware NIC.
+     *
+     * IPS mode -- with one endpoint a host stack "NIC":
+     * When using multiple rings/threads are being used, then the open of the initial Ring 0 MUST
```

Comment:
The clause in the first line of this comment has some extra words. It should say "_When using multiple rings/threads, ..._" instead of "_When using multiple rings/threads are being used, ..._".